### PR TITLE
Hw20 andriy povzun

### PR DESCRIPTION
**Pull request by AndriyPovzun**

### CODE REVIEW

File: Andriy_Povzun/HW20/conftest.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    return Human(123, random.randint(0, 100), random.choice(gender))
+
+
+@pytest.fixture
+def create_human_negative_age(faker):
+    return Human(faker.name(), -55, random.choice(gender))
+
+
+@pytest.fixture
+def human_grow_to_100_years(create_valid_human):
```
Comment (bohdandovbsyh) [inline Andriy_Povzun/HW20/conftest.py]: it makes no sense. If you need Human with 100 years just create it

File: Andriy_Povzun/HW20/test_human/test_human_class.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    with pytest.raises(Exception):
+        human = create_human_negative_age
+
+
+@pytest.mark.negative
+def test_error_when_set_same_gender(create_valid_human):
+    human = create_valid_human
+    gender = human.gender
+    with pytest.raises(Exception):
+        human.change_gender(gender)
```
Comment (bohdandovbsyh) [inline Andriy_Povzun/HW20/test_human/test_human_class.py:21]: you can test error message here as well

File: Andriy_Povzun/HW20/test_human/test_human_class.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+
+@pytest.mark.negative
+def test_change_gender_on_invalid(create_valid_human):
+    human = create_valid_human
+    with pytest.raises(Exception):
+        human.change_gender('bla')
```
Comment (bohdandovbsyh) [inline Andriy_Povzun/HW20/test_human/test_human_class.py]: do not hardcode data. Use random string for example

File: Andriy_Povzun/HW20/test_human/test_human_class.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+
+@pytest.mark.negative
+def test_human_grow_after_100_year(human_grow_to_100_years):
+    human = human_grow_to_100_years
+    human.grow()
+    assert human.age == 100
```
Comment (bohdandovbsyh) [inline Andriy_Povzun/HW20/test_human/test_human_class.py]: this assert is not required

File: Andriy_Povzun/HW20/test_human/test_human_class.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+@pytest.mark.negative
+def test_change_gender_on_invalid(create_valid_human):
+    human = create_valid_human
+    with pytest.raises(Exception):
+        human.change_gender('bla')
+
+
+@pytest.mark.negative
+def test_human_grow_after_100_year(human_grow_to_100_years):
+    human = human_grow_to_100_years
```
Comment (bohdandovbsyh) [inline Andriy_Povzun/HW20/test_human/test_human_class.py]: just create human with age 100

File: Andriy_Povzun/HW20/test_human/test_human_class.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    human.grow()
+    assert human.age == 100
+    with pytest.raises(Exception):
+        human.grow()
```
Comment (bohdandovbsyh) [inline Andriy_Povzun/HW20/test_human/test_human_class.py:36]: please check error message as well

File: Andriy_Povzun/HW20/test_human/test_human_class.py [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+def test_success_change_human_gender(create_valid_human):
+    male = 'male'
+    female = 'female'
+    human = create_valid_human
+    if human.gender == male:
+        human.change_gender(female)
+        assert human.gender == female
+    else:
+        human.change_gender(male)
+        assert human.gender == male
```
Comment (bohdandovbsyh) [inline Andriy_Povzun/HW20/test_human/test_human_class.py]: if/else is a VERY bad practice in test cases. Use parametrization instead

File: Andriy_Povzun/HW20/test_human/test_human_class.py [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    with pytest.raises(Exception):
+        human.grow()
+
+
+@pytest.mark.smoke
+def test_check_grow_func(create_valid_human):
+    human = create_valid_human
+    age_before_func = human.age
+    human.grow()
+    assert human.age == age_before_func + 1
```
Comment (bohdandovbsyh) [inline Andriy_Povzun/HW20/test_human/test_human_class.py]: please add assert message

Comment (bohdandovbsyh) [inline Andriy_Povzun/HW20/test_human/test_human_class.py]: General comment. I do not see positive testing. For example :  test_age, test_gender, death....Please think about more TCs

**Review by bohdandovbsyh — APPROVED** [review]

File: Andriy_Povzun/HW20/test_human/test_human_class.py (hunk 7 above)
Comment (AndriyPovzun) [reply Andriy_Povzun/HW20/test_human/test_human_class.py]: okay, fixed

File: Andriy_Povzun/HW20/test_human/test_human_class.py (hunk 8 above)
Comment (AndriyPovzun) [reply Andriy_Povzun/HW20/test_human/test_human_class.py]: done

Comment (AndriyPovzun) [reply Andriy_Povzun/HW20/test_human/test_human_class.py]: fixed.

File: Andriy_Povzun/HW20/test_human/test_human_class.py (hunk 6 above)
Comment (AndriyPovzun) [reply Andriy_Povzun/HW20/test_human/test_human_class.py:36]: done

File: Andriy_Povzun/HW20/test_human/test_human_class.py (hunk 5 above)
Comment (AndriyPovzun) [reply Andriy_Povzun/HW20/test_human/test_human_class.py]: done

File: Andriy_Povzun/HW20/test_human/test_human_class.py (hunk 4 above)
Comment (AndriyPovzun) [reply Andriy_Povzun/HW20/test_human/test_human_class.py]: this assert was removed

File: Andriy_Povzun/HW20/test_human/test_human_class.py (hunk 3 above)
Comment (AndriyPovzun) [reply Andriy_Povzun/HW20/test_human/test_human_class.py]: Understood, done

File: Andriy_Povzun/HW20/test_human/test_human_class.py (hunk 2 above)
Comment (AndriyPovzun) [reply Andriy_Povzun/HW20/test_human/test_human_class.py:21]: done

File: Andriy_Povzun/HW20/conftest.py (hunk 1 above)
Comment (AndriyPovzun) [reply Andriy_Povzun/HW20/conftest.py]: done

**Review by bohdandovbsyh — APPROVED** [review]

**Pull request merged**